db/migrate/20170925210519_add_last_existed_in_storage_root_and_last_existed_in_db_and_last_checksum_validation_to_preservation_copies.rb

```diff
@@ -0,0 +1,7 @@
+class AddLastExistedInStorageRootAndLastExistedInDbAndLastChecksumValidationToPreservationCopies < ActiveRecord::Migration[5.1]
+  def change
+    add_column :preservation_copies, :last_existed_in_storage_root, :datetime
+    add_column :preservation_copies, :last_existed_in_db, :datetime
```

Comment:
i would suggest maybe `last_checked_on_storage` and `last_checked_in_catalog`, respectively.  the bit of nuance being, in an error case, i.e. if the thing we expected isn't on disk at the version we expect, or it isn't already in the catalog, that column will indicate the last time we updated our knowledge of that state, but it won't necessarily indicate that the thing was in the place we expected at that time (because maybe it wasn't).

unless we stop updating those fields when we notice something's amiss, in which case those would be accurate names.

one more stray thought:  maybe we don't actually need `last_existed_in_db`/`last_checked_in_catalog`/whatever.  maybe that's implied by the `updated_at` timestamp.

and yet another stray thought, though it sort of ties into all of the above:  maybe `last_existed_in_db` is inherently non-sensical, because if the row did somehow disappear from the DB after being there for a time, we've already lost that information.  `last_checked_in_catalog` seems slightly less semantically sticky in that regard, but i'm also not sure it's any more useful than `updated_at` in light of that scenario.

so i dunno, maybe i'm actually just arguing to change the name of `last_existed_in_storage_root` and to drop `last_existed_in_db`.  i'm not entirely sure what the changes i'm looking for here are, obviously 😄.